src/consul_configs.rs

```diff
@@ -73,3 +74,26 @@ where
         }
     }
 }
+
+pub fn watch_for_config_deletion (
+    chain: &str,
+    consul_url: &str,
+    consul_token: &str,
+    community: &str,
+) {
+    let client = Client::new(consul_url, consul_token);
+    let keystore = client.keystore;
+    let one_sec = time::Duration::from_secs(1);
+
+    loop {
+        let keyname = format!("chain/{}/{}", &community, &chain);
+
+        if let Ok(_) = keystore.get_key(keyname) {
```

Comment:
Is there a race here or will rust consul somehow report if the modification index changes? Seems like if the key is deleted then recreated in <1s this will fail

Reply:
Yes, there would be. See update